File: algorithms/10_electric_energy.c
#include <stdio.h>

void energyValue(double kwh, char type_of_installation);

int main()
{
    double kwh;
    char type_of_installation;

    printf("Qual a quantidade de Kwh consumida? ");
    scanf("%lf", &kwh);

    printf("Qual o tipo de instalação? (R para residências, I para indústrias e C para comércios)\n");
    scanf("%s", &type_of_installation);

    energyValue(kwh, type_of_installation);

    return 0;
}

void energyValue(double kwh, char type_of_installation)
{

    double value;

    switch (type_of_installation)
    {
    case 'R':
        if (kwh <= 500)
        {
            value = kwh * 0.4;
        }
        else
        {
            value = kwh * 0.65;
        }

        printf("O valor da energia é de R$%.2lf\n", value);
        break;
    case 'C':
        if (kwh <= 1000)
        {
            value = kwh * 0.55;
        }
        else
        {
            value = kwh * 0.6;
        }

        printf("O valor da energia é de R$%.2lf\n", value);
        break;
    case 'I':
        if (kwh <= 5000)
        {
            value = kwh * 0.55;
        }
        else
        {
            value = kwh * 0.6;
        }

        printf("O valor da energia é de R$%.2lf\n", value);
        break;

    default:
        printf("Operação inválida");
        break;
    }
}

File: algorithms/10_electric_energy.py
def energyValue(kwh, type_of_installation):

    if type_of_installation == 'R':
        if kwh <= 500:
            return kwh * 0.4
        else:
            return kwh * 0.65
    elif type_of_installation == 'C':
        if kwh <= 1000:
            return kwh * 0.55
        else:
            return kwh * 0.6

    elif type_of_installation == 'I':
        if kwh <= 5000:
            return kwh * 0.55
        else:
            return kwh * 0.6
    else:
        print("Operação inválida")
        exit()


kwh = float(input("Qual a quantidade de Kwh consumida? "))
type_of_installation = str(
    input("Qual o tipo de instalação? (R para residências, I para indústrias e C para comércios)\n")
)

print("O valor da energia é de R$%.2lf" % energyValue(kwh, type_of_installation))
